content/blog/using-lua-as-esp-idf-component-with-esp32/index.md

```diff
@@ -0,0 +1,187 @@
+        ESP_LOGE(TAG, "Failed to set package.path: %s", lua_tostring(L, -1));
+        lua_pop(L, 1);
+    }
+
+    // Load and execute the Lua script from file
+    if (luaL_dofile(L, file_name) != LUA_OK) {
+        ESP_LOGE(TAG, "Error executing Lua script '%s': %s", file_name, lua_tostring(L, -1));
+        lua_pop(L, 1);
+    }
+
+    lua_close(L);
+    ESP_LOGI(TAG, "Finished running Lua script: %s", description);
+}
+```
+
+Creates a new Lua state and opens standard libraries.
+Sets the package.path to include the filesystem path where Lua scripts are stored.
+Executes the Lua script from the specified file.
+
+### Filesystem Initialization
+
+Function: `init_spiffs`
+
+```c
+void init_spiffs(void) {
+    ESP_LOGI(TAG, "Initializing SPIFFS");
+
+    esp_vfs_spiffs_conf_t conf = {
+        .base_path = "/spiffs",
+        .partition_label = NULL,
+        .max_files = 5,
+        .format_if_mount_failed = true
+    };
+
+    esp_err_t ret = esp_vfs_spiffs_register(&conf);
+
+    if (ret != ESP_OK) {
+        ESP_LOGE(TAG, "Failed to mount or format filesystem");
+        return;
+    }
+
+    size_t total = 0, used = 0;
+    ret = esp_spiffs_info(NULL, &total, &used);
+    if (ret == ESP_OK) {
+        ESP_LOGI(TAG, "SPIFFS total: %d, used: %d", total, used);
+    } else {
+        ESP_LOGE(TAG, "Failed to get SPIFFS partition information");
+    }
+}
+```
+
+
+### Getting Started
+
+Clone the Repository
+
+```bash
+git clone https://github.com/yourusername/esp32-lua-example.git
+cd esp32-lua-example
+```
+
+Configure the Project
+
+Use idf.py menuconfig to configure the project settings if necessary.
+
+Build, flash and monitor
+
+```bash
+idf.py build flash monior
+```
+
+Expected output:
+
+```shell
+The answer is: 42
+Fibonacci of 10 is: 55
+QR Code for: https://developer.espressif.com/tags/lua
+```
+
+
+## Integrating Lua as an ESP-IDF Component
+
+To incorporate Lua into your ESP32 project seamlessly, you can use the ESP-IDF Component Registry, which simplifies dependency management and ensures you're using a stable version of Lua.
+
+### Adding Lua to Your Project
+In your project's main directory, create or edit the idf_component.yml file to specify Lua as a dependency:
+
+```yaml
+# main/idf_component.yml
+
+dependencies:
+  georgik/lua: "==5.4.7"
+  joltwallet/littlefs: "==1.14.8"
+
+idf:
+  version: ">=5.0.0"
+```
+This configuration tells the ESP-IDF build system to include version 5.4.7 of the Lua component maintained by georgik, as well as version 1.14.8 of the LittleFS component for filesystem support. The idf section ensures compatibility with ESP-IDF version 5.0.0 or newer.
+
+### Utilizing the ESP-IDF Component Registry
+
+By specifying Lua in your idf_component.yml, the ESP-IDF build system automatically downloads and integrates the Lua component during the build process. There's no need to manually clone or manage the Lua source code.
```

Comment:
Should we consider using monospace for file paths and names?

```suggestion
By specifying Lua in your `idf_component.yml`, the ESP-IDF build system automatically downloads and integrates the Lua component during the build process. There's no need to manually clone or manage the Lua source code.
```

If yes, suggest updating the other file paths in names in this article.